FirebaseCombineSwift/Sources/Firestore/Transaction+Combine.swift

```diff
@@ -0,0 +1,80 @@
+
+@available(swift 5.0)
+@available(macOS 10.15, iOS 13, watchOS 6, tvOS 13, *)
+extension Firestore {
+    
+    /// Executes the given updateBlock and then attempts to commit the changes applied within an
+    /// atomic transaction.
+    ///
+    /// The maximum number of writes allowed in a single transaction is 500, but note that each
+    /// usage of `FieldValue.serverTimestamp()`, `FieldValue.arrayUnion()`,
+    /// `FieldValue.arrayRemove()`, or `FieldValue.increment()` inside a transaction counts as an
+    ///  additional write.
+    ///
+    /// In the updateBlock, a set of reads and writes can be performed atomically using the
+    ///  `Transaction` object passed to the block. After the updateBlock is run, Firestore will
+    ///  attempt to apply the changes to the server. If any of the data read has been modified
+    ///  outside of this transaction since being read, then the transaction will be retried by
+    ///  executing the updateBlock again. If the transaction still fails after 5 retries, then the
+    ///   transaction will fail.
+    ///
+    /// Since the updateBlock may be executed multiple times, it should avoiding doing anything that
+    /// would cause side effects.
+    ///
+    /// Any value maybe be returned from the updateBlock. If the transaction is successfully
+    /// committed, then the completion block will be passed that value. The updateBlock also has an
+    ///  `Error` out parameter. If this is set, then the transaction will not attempt to commit, and
+    ///  the given error will be passed to the completion block.
+    ///
+    /// The `Transaction` object passed to the updateBlock contains methods for accessing documents
+    ///  and collections. Unlike other firestore access, data accessed with the transaction will not
+    ///  reflect local changes that have not been committed. For this reason, it is required that
+    ///  all reads are performed before any writes. Transactions must be performed while online.
+    ///  Otherwise, reads will fail, the final commit will fail, and the completion block will
+    ///  return an error.
+    ///
+    /// - Parameter updateBlock: The block to execute within the transaction context.
+    /// - Returns: A publisher emitting a value instance passed from the updateBlock. This block
+    ///  will run even if the client is offline, unless the process is killed.
+    public func runTransaction<T>(_ updateBlock: @escaping (Transaction) throws -> T)
+    -> Future<T, Error> {
+        Future { promise in
+            self.runTransaction({ transaction, errorPointer in
+                do {
+                    return try updateBlock(transaction)
+                } catch {
+                    errorPointer?.pointee = error as NSError
+                    return nil
+                }
+            }) { value, error in
+                if let error = error {
+                    promise(.failure(error))
+                } else if let value = value as? T {
```

Comment:
Not sure what happens if both `error` and `value` are nil.

@schmidt-sebastian can you advise?

Reply:
To better understand this problem.
`nil`s can be optional:
```swift
var t1: Int? = nil
var t2: Int?? = t1

if let t = t2 {
    print("OK", t) // This prints
}
```
Or better:
```swift
var t: Int?? = .some(nil)
if let t = t {
    print("OK", t) // This Prints
}
```
```swift
var t: Int?? = nil
if let t = t {
    print("OK", t) // This doesn't print
}
```